src/pages/iv-items/forms/SalesForm.js

```diff
@@ -0,0 +1,250 @@
+import { useForm } from 'src/hooks/form'
+import { useInvalidate } from 'src/hooks/resource'
+import { ControlContext } from 'src/providers/ControlContext'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { DataSets } from 'src/resources/DataSets'
+
+const SalesForm = ({ labels, maxAccess, store, record, cId }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.Sales.qry
+  })
+
+  const { recordId: itemId } = store
+
+  const validationSchema = yup.object({})
+
+  const { formik } = useForm({
+    maxAccess,
+    initialValues: {
+      itemId,
+      currencyId: cId,
+      plId: '',
+      priceType: '',
+      valueType: '',
+      value: '',
+      priceWithVat: '',
+      minPrice: '',
+
+      ...record
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      currencyId: yup.string().required(),
+      plId: yup.string().required(),
+      priceType: yup.string().required(),
+      value: yup.string().required(),
+      valueType: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      const plId = formik.values.plId
+      const currencyId = formik.values.currencyId
+
+      const response = await postRequest({
+        extension: SaleRepository.Sales.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!plId && !currencyId) {
+        toast.success(platformLabels.Added)
+      } else toast.success(platformLabels.Edited)
+
+      formik.setValues(obj)
+
+      invalidate()
+    }
+  })
+  const editMode = !!record
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (record && record.plId) {
+        try {
+          const res = await getRequest({
+            extension: SaleRepository.Sales.get,
+            parameters: `_itemId=${itemId}&_plId=${formik.values.plId}&_currencyId=${formik.values.currencyId}`
+          })
+
+          if (res.record) {
+            formik.setValues(res.record)
+          }
+        } catch (error) {}
+      }
+    }
+
+    fetchData()
+  }, [record])
+
+  useEffect(() => {
+    if (!editMode) {
+      ;(async () => {
+        const responsePriceLevel = await getRequest({
+          extension: SaleRepository.PriceLevel.qry,
+          parameters: `_filter=`
+        })
+        if (responsePriceLevel.list && responsePriceLevel.list.length > 0) {
+          const firstPriceLevel = responsePriceLevel.list[0]
+          formik.setFieldValue('plId', firstPriceLevel.recordId)
+        }
+
+        const responsePriceType = await getRequest({
+          extension: InventoryRepository.Items.pack
+        })
+        if (responsePriceType.record && responsePriceType.record.priceTypes.length > 0) {
```

Comment:
this if block can be replaced by 
formik.setFieldValue('priceType', parseInt(responsePriceType?.record?.priceTypes[0]?.key))